contracts/BullaFinance.sol

```diff
@@ -116,14 +171,56 @@ abstract contract BullaFinance {
     ///     Allows a debtor to accept a creditor's financing offer and begin payment
     ///     This function will:
     ///         RES1. load the previous claim details and create a new bulla claim specifying `claimAmount` as `originatingClaimAmount + (originatingClaimAmount * terms.interestBPS / 10_000)` and `dueBy` as `term.termLength + block.timestamp`
-    ///         RES2. deletes the `financeTerms` TODO: does this have any potential negative side-effects / drawbacks?
-    ///         RES3. increments `withdrawableFee` by `fee`
+    ///         RES2. deletes the `financeTerms`
+    ///         RES3. increments `unharvestedTransactions`
     ///         RES4. pays `downPayment` amount on the newly created claim
     ///         RES5. emits a LoanAccepted event with the `originatingClaimId` and the new claimId as `financedClaimId`
+    ///         RETURNS: the newly created claimId
     ///     Given the following:
     ///         P1. msg.sender has approved BullaFinance to spend at least `downPayment` amount of the underlying claim's denominating ERC20 token
     ///         P2. `financingTerms[claimId].termLength != 0` (offer exists)
-    ///         P3. `msg.sender == claim.debtor`
-    ///         P4. `downPayment >= (claimAmount * minDownPaymentBPS / 10_000)` && `downPayment < claimAmount + (claimAmount * minDownPaymentBPS / 10_000) (not overpaying or underpaying)
-    function acceptLoan(uint256 claimId, uint256 downPayment) public virtual;
+    ///         P3. `claim.status == ClaimStatus.Pending`
+    ///         P4. `msg.sender == claim.debtor`
+    ///         P5. `downPayment >= (claimAmount * minDownPaymentBPS / 10_000)` && `downPayment < claimAmount + (claimAmount * interestBPS / 10_000) (not overpaying or underpaying)
+    function acceptFinancing(
+        uint256 claimId,
+        uint256 downPayment,
+        string memory newDescription
```

Comment:
not sure about this newDescription. Have to think about it

Reply:
It's just that description and tags are stored off chain - so they need to be resupplied as parameters